packages/types/src/client.ts

```diff
@@ -291,6 +292,24 @@ export interface Client<O extends ClientOptions = ClientOptions> {
    */
   on(hook: 'startNavigationSpan', callback: (options: StartSpanOptions) => void): () => void;
 
+  /**
+   * A hook for GraphQL client integration to enhance a span with request data.
+   * @returns A function that, when executed, removes the registered callback.
+   */
+  on(
+    hook: 'beforeOutgoingRequestSpan',
+    callback: (span: Span, handlerData: HandlerDataXhr | HandlerDataFetch) => void,
```

Comment:
m: Let's instead type this like this:

```suggestion
    callback: (span: Span, handlerData: FetchBreadcrumbHint | XhrBreadcrumbHint) => void,
```

So give it the (typed) hint instead. this should be a bit more "stable" then the handler data, and the relevant handler data should be part of the hint instead?